src/Groups/matrices/stuff_field_gen.jl

```diff
@@ -0,0 +1,52 @@
+import Hecke: evaluate, field_extension, FinField, FinFieldElem, PolyElem
+
+export primitive_element
+
+function _change_type(f::PolyElem{T}) where T <: FinFieldElem
+   e,p = ispower(order(base_ring(f)))
+   F = GF(Int(p),Int(e))[1]
+   t = PolynomialRing(F,"t")[2]
+   return sum([t^i*F(lift(coeff(f,i))) for i in 0:degree(f)])
+end
+
+# return a generator for the unit group of F
+function _centralizer(f::PolyElem{T}) where T <: FinFieldElem
+  if typeof(f)!=fq_nmod_poly && typeof(f)!=fq_poly
+     f = _change_type(f)
+  end
+  K = base_ring(f)
+  d = degree(K)
+  e = degree(f)
+  q = order(K)
```

Comment:
These seem to be unused?
```suggestion
```

Reply:
`d`, `e`, `q` for sure. I just leave `K` commented in the case we want to resume the other two commented lines (the result of "uncomment" these lines is to produce a generator for `F* / (prime subfield)*` .